asf_heat_pump_suitability/getters/get_dno_datasets.py

```diff
@@ -0,0 +1,467 @@
+import re
+from typing import Any
+import logging
+
+import numpy as np
+import pandas as pd
+import geopandas as gpd
+
+from asf_heat_pump_suitability import config
+from asf_heat_pump_suitability.getters import base_getters
+from asf_heat_pump_suitability.getters.s3_getters import (
+    load_s3_data,
+)
+from asf_heat_pump_suitability.utils.geo_utils import parse_binary_geometry
+
+
+CRS = "EPSG:4326"  # Geometry coordinate reference system - standard longitude/latitude projection
+
+
+def generate_enw_gdf() -> gpd.GeoDataFrame:
+    """
+    Generate a GeoDataFrame for Electricity North West (ENW) primary substations and their service areas.
+
+    The function processes substation data from ENW's Distribution Future Energy Scenarios (DFES)
+    and Network Development Plan (NDP) datasets. It combines point locations of substations with
+    their corresponding service area polygons (Voronoi polygons) and capacity/demand information.
+
+    Returns:
+        gpd.GeoDataFrame: DataFrame with EPSG:4326 (WGS84) projection containing:
+            - id: Unique substation identifier
+            - firm_capacity_mva: Maximum power the substation can safely deliver (MVA)
+            - peak_demand_mva: Maximum observed power demand at the substation (MVA)
+            - geo_shape: Polygon geometry representing the substation's service area
+            - operator: Distribution network operator code ("ENW")
+    """
+    enw_demand = base_getters.get_df_from_parquet_s3_path(
+        config["data_source"]["ENW_dfes_primaries"]
+    ).to_pandas()
+    enw_substations = base_getters.get_df_from_parquet_s3_path(
+        config["data_source"]["ENW_ndp_headroom"]
+    ).to_pandas()
+    enw_shape = base_getters.get_df_from_parquet_s3_path(
+        config["data_source"]["ENW_ndp_voronoi"]
+    ).to_pandas()
+
+    # Filter data for primary substations, current year, and best view scenario
```

Comment:
```suggestion
    # Filter data for primary substations, current year, and best view scenario
    # Best view scenario represents middle ground future projections of demand on substation
```
Feel free to correct it as appropriate but something like this will help me understand the code later!